meson.build

```diff
@@ -768,71 +737,68 @@ uchardet_opt = get_option('uchardet').require(
 uchardet = dependency('uchardet', required: uchardet_opt)
 if uchardet.found()
     dependencies += uchardet
-    features += 'uchardet'
 endif
+features += {'uchardet': uchardet.found()}
 
 vapoursynth = dependency('vapoursynth', version: '>= 24', required: get_option('vapoursynth'))
 vapoursynth_script = dependency('vapoursynth-script', version: '>= 23',
                                 required: get_option('vapoursynth'))
 if vapoursynth.found() and vapoursynth_script.found()
     dependencies += [vapoursynth, vapoursynth_script]
-    features += 'vapoursynth'
     sources += files('video/filter/vf_vapoursynth.c')
 endif
+features += {'vapoursynth': vapoursynth.found() and vapoursynth_script.found()}
```

Comment:
super smart idea I had just now, reorder the lines so it can be like this:
```meson
features += {'vapoursynth': vapoursynth.found() and vapoursynth_script.found()}
if features['vapoursynth']
    dependencies += [vapoursynth, vapoursynth_script]
    # ...
endif
```